src/road_cmd.cpp

```diff
@@ -1929,15 +1929,15 @@ static Foundation GetFoundation_Road(TileIndex tile, Slope tileh)
 	}
 }
 
-static const Roadside _town_road_types[][2] = {
+static const Roadside _town_road_types[HZB_END][2] = {
```

Comment:
Not necessarily bad, but there's a tiny caveat in the specifications IIRC. If we make `HZB_END` bigger, it will just fill the remainder with default initialized values. So, maybe leaving `HZB_END` out of this, but adding `static_assert(lengthof(_town_road_types) == HZB_END);` after the table would be safer?
For now I would not immediately update the PR, but rather await some consensus on this.

Reply:
^^ I agree with this.